zebra-network/src/peer_set/set.rs

```diff
@@ -250,12 +259,38 @@ where
         let (_, svc) = self.ready_services.get_index(index).expect("invalid index");
         svc.load()
     }
+
+    fn best_peer_for(&mut self, req: &Request) -> (SocketAddr, D::Service) {
+        if let Request::BlocksByHash(hashes) = req {
+            for hash in hashes.iter() {
+                let mut peers = self.inventory_registry.peers(&(*hash).into());
```

Comment:
Why can't this be `peers(hash)`? From looking at the code I expect that `hash: &block::Hash`, so I'm surprised that an into conversion is needed.

Reply:
because `peers()` takes a `InventoryHash` instead of a `block::Hash`.

Reply:
ah, got it.